mithril-test-lab/mithril-aggregator-fake/scripts/import.sh

```diff
@@ -142,11 +149,35 @@ EOF
         let "nb=nb+1"
         echo -n "$nb   "
     done
-    echo "]" >> $DATA_DIR/ctx-proofs.json
+    echo "]" >> $DATA_DIR/ctx-proofs-list.json
 
     echo " ✅";
 }
 
+# Join downloaded artifacts files into a single files
+# $1=name of the artifacts to join
+join_artifacts_files() {
+    local -r name=${1:?"No artifact name given to join_artifacts_files function."};
+    local -r src="${DATA_DIR:?}/${name}"
+    local -r dest="$DATA_DIR"/"$name"s.json
+    local buffer=""
+    local separator=""
+
+    echo "Joining ${name} artifacts into ${dest} …"
+
+    buffer+="{"
+    for filename in "$src/"*; do
+        local key=$(basename "${filename%.*}")
+
+        buffer+=$(echo "$separator\"$key\": " | cat - "$filename")
+        separator=","
+    done
+    buffer+="}"
```

Comment:
You could also have created the JSON file with `jq` here.

Reply:
We don't know how to do that with pure jq (the problem is on how to construct an hashmap,), do you have an idea ?

Reply:
I'd say that with `--argfile` argument it should work:
```bash
// Write value file
$ echo "{}" | jq --argjson val 123456 '. += {"test_field_1": $val}' > test.json
$ cat test.json
{
  "test_field_1": 123456
}

// Insert file content as a value for a new key
$ echo "{}" | jq --argfile val test.json '. += {"test_field_2": $val}'
{
  "test_field_2": {
    "test_field_1": 123456
  }
}
```